Tests/xcodeprojMeasurementsTests/XcodeProjGenerator.swift

```diff
@@ -0,0 +1,144 @@
+import Foundation
+import PathKit
+@testable import xcodeproj
+
+final class XcodeProjGenerator {
+    
+    struct GenerateSetting {
+        let targetCount: Int = 5
+        let groupCountInLevel: Int = 4
+        let fileCountInLevel: Int = 10
+        let maxGroupDepth: Int = 4
+    }
+    
+    @discardableResult
+    func generate(path: Path, sourceRoot: Path, setting: GenerateSetting = GenerateSetting()) throws -> XcodeProj {
+        let workspace = XCWorkspace()
+        let buildConfigurationList = XCConfigurationList()
+        let pbxproj = PBXProj(rootObject: nil, objectVersion: 0, archiveVersion: 0, classes: [:], objects: [])
+        let rootGroupName = UUID().uuidString
+        let rootGroupPath = sourceRoot + Path(rootGroupName)
+        try rootGroupPath.mkpath()
+        let mainGroup = PBXGroup(
+            children: [],
+            sourceTree: .group,
+            name: rootGroupName,
+            path: rootGroupName
+        )
+        pbxproj.add(object: mainGroup)
+        let project = PBXProject(
+            name: UUID().uuidString,
+            buildConfigurationList: buildConfigurationList,
+            compatibilityVersion: "Xcode 9.3",
+            mainGroup: mainGroup
+        )
+        pbxproj.add(object: project)
+        pbxproj.rootObject = project
+        let xcodeproj = XcodeProj(workspace: workspace, pbxproj: pbxproj, sharedData: nil)
+        xcodeproj.pbxproj.add(object: buildConfigurationList)
+        try fillProject(
+            pbxproj: xcodeproj.pbxproj,
+            rootProject: project,
+            sourceRoot: sourceRoot,
+            mainGroup: mainGroup,
+            setting: setting
+        )
+        try xcodeproj.write(path: path)
+        return xcodeproj
+    }
+    
+    private func fillProject(
+        pbxproj: PBXProj,
+        rootProject: PBXProject,
+        sourceRoot: Path,
+        mainGroup: PBXGroup,
+        setting: GenerateSetting) throws
+    {
+        try (0..<setting.targetCount).forEach { _ in
+            let target = generateTargets(pbxproj: pbxproj)
+            rootProject.targets.append(target)
+            var buildFiles = [PBXBuildFile]()
+            let currentPath = try mainGroup.fullPath(sourceRoot: sourceRoot)
```

Comment:
I'd calculate the path based on how the project is being generated instead of using those convenient methods.